demos/gaze_estimation_demo/src/eye_state_estimator.cpp

```diff
@@ -50,41 +50,43 @@ void EyeStateEstimator::rotateImageAroundCenter(const cv::Mat& srcImage,
 }
 
 void EyeStateEstimator::estimate(const cv::Mat& image, FaceInferenceResults& outputResults) {
+    outputResults.leftEyeMidpoint = (outputResults.faceLandmarks[0] + outputResults.faceLandmarks[1]) / 2;
+    outputResults.rightEyeMidpoint = (outputResults.faceLandmarks[2] + outputResults.faceLandmarks[3]) / 2;
     auto roll = outputResults.headPoseAngles.z;
 
-    auto leftEyeBoundingBox = createEyeBoundingBox(outputResults.faceLandmarks[0],
-        outputResults.faceLandmarks[1]);
-    auto rightEyeBoundingBox = createEyeBoundingBox(outputResults.faceLandmarks[2],
-        outputResults.faceLandmarks[3]);
-    auto leftEyeImage(cv::Mat(image, leftEyeBoundingBox));
-    auto rightEyeImage(cv::Mat(image, rightEyeBoundingBox));
-
-    cv::Mat leftEyeImageRotated, rightEyeImageRotated;
-    rotateImageAroundCenter(leftEyeImage, leftEyeImageRotated, roll);
-    rotateImageAroundCenter(rightEyeImage, rightEyeImageRotated, roll);
-    leftEyeImage = leftEyeImageRotated;
-    rightEyeImage = rightEyeImageRotated;
-
+    auto leftEyeBoundingBox = createEyeBoundingBox(outputResults.faceLandmarks[0], outputResults.faceLandmarks[1]);
     outputResults.leftEyeBoundingBox = leftEyeBoundingBox;
+    if (outputResults.faceLandmarks[0] != outputResults.faceLandmarks[1]) {
```

Comment:
Maybe it would be better to check `leftEyeBoundingBox` here instead of the original landmarks? It is `leftEyeBoundingBox` that's used as the ROI.

Reply:
Ok, but don't see a difference as there are comments pointing out the linkage.

Reply:
The more obvious the connection, the better.

Reply:
Well, `outputResults.faceLandmarks[0] != outputResults.faceLandmarks[1]` suggests there is a connection, while checking `leftEyeBoundingBox` doesn't :)

Reply:
I mean a connection between the condition in the `if` statement and the body. :-)